File: roadhelp_web/src/main/java/com/codedy/roadhelp/model/Issues.java
package com.codedy.roadhelp.model;

import com.codedy.roadhelp.model.enums.IssueCategory;
import com.codedy.roadhelp.model.enums.IssueStatus;

import javax.persistence.*;
import javax.validation.constraints.Size;
import java.io.Serializable;

@Entity
@Table(name = "issue")
public class Issues extends BaseModel implements Serializable {

    //region - Define Fields -
    @Size(max = 128)
    private String address;

    @Size(min = 10)
    private String phone;

    private double longitude;

    private double latitude;

    private IssueCategory category;

    @Size(max = 500)
    private String description;

    private IssueStatus status;
    //endregion


    //region - Relationship -
    @ManyToOne
    @JoinColumn(name = "user_member_id")
    private User userMember;

    @ManyToOne
    @JoinColumn(name = "user_partner_id")
    private User userPartner;

    @OneToOne(mappedBy = "issues")
    private RatingIssues ratingIssues;
    //endregion


    //region - Getter & Setter -
    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public IssueCategory getCategory() {
        return category;
    }

    public void setCategory(IssueCategory category) {
        this.category = category;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public IssueStatus getStatus() {
        return status;
    }

    public void setStatus(IssueStatus status) {
        this.status = status;
    }

    public User getUserMember() {
        return userMember;
    }

    public void setUserMember(User userMember) {
        this.userMember = userMember;
    }

    public User getUserPartner() {
        return userPartner;
    }

    public void setUserPartner(User userPartner) {
        this.userPartner = userPartner;
    }

    public RatingIssues getRatingIssues() {
        return ratingIssues;
    }

    public void setRatingIssues(RatingIssues ratingIssues) {
        this.ratingIssues = ratingIssues;
    }
    //endregion


    @Override
    public String toString() {
        return "Issues{" +
                "address='" + address + '\'' +
                ", phone='" + phone + '\'' +
                ", longitude=" + longitude +
                ", latitude=" + latitude +
                ", category=" + category +
                ", description='" + description + '\'' +
                ", status=" + status +
                ", userMember=" + userMember +
                ", userPartner=" + userPartner +
                ", ratingIssues=" + ratingIssues +
                '}';
    }

}

File: roadhelp_app/lib/models/issues.dart
import 'dart:convert';

import '/config/enums.dart';
import '/models/rating_issues.dart';
import '/models/user.dart';
import 'base_model.dart';

class Issues extends BaseModel {
  //
  int? userMemberId;
  int? userPartnerId;

  //
  String? address;
  String? phone;
  double? longitude;
  double? latitude;
  IssueCategory? category;
  String? description;
  IssueStatus? status;

  //Relationship
  User? userMember;
  User? userPartner;
  RatingIssues? ratingIssue;

  Issues({
    this.userMemberId,
    this.userPartnerId,
    this.address,
    this.phone,
    this.longitude,
    this.latitude,
    this.category,
    this.description,
    this.status,
    //Relationship
    this.userMember,
    this.userPartner,
    this.ratingIssue,
    //
    int? id,
    DateTime? createdAt,
    String? createdBy,
    DateTime? updatedAt,
    String? updatedBy,
    int? version,
    bool? deleted,
  }) : super(
          id: id,
          createdAt: createdAt,
          createdBy: createdBy,
          updatedAt: updatedAt,
          updatedBy: updatedBy,
          version: version,
          deleted: deleted,
        );

  //
  factory Issues.fromJson(Map<String, dynamic> json) {
    return Issues(
      userMemberId: json['userMemberId'],
      userPartnerId: json['userPartnerId'],
      //
      longitude: json['longitude'],
      latitude: json['latitude'],
      phone: json['phone'],
      address: json['address'],
      category: IssueCategory.values.firstWhere(
          (e) => e.toString() == "IssueCategory." + json['category']),
      description: json['description'],
      status: IssueStatus.values
          .firstWhere((e) => e.toString() == "IssueStatus." + json['status']),
      //Relationship
      userMember:
          json['userMember'] != null ? User.fromJson(json['userMember']) : null,
      userPartner: json['userPartner'] != null
          ? User.fromJson(json['userPartner'])
          : null,
      ratingIssue: json['ratingIssue'] != null
          ? RatingIssues.fromJson(json['ratingIssue'])
          : null,
      //
      id: json['id'],
      //createdAt: json['createdAt'],
      createdBy: json['createdBy'],
      //updatedAt: json['updatedAt'],
      updatedBy: json['updatedBy'],
      version: json['version'],
      deleted: json['deleted'],
    );
  }

  //
  static List<Issues> fromJsonToList(dynamic json) {
    return json.map<Issues>((element) => Issues.fromJson(element)).toList();
  }

  //
  String toJson() {
    return json.encode({
      'userMemberId': userMemberId,
      'userPartnerId': userPartnerId,
      //
      'address': address,
      'phone': phone,
      'longitude': longitude,
      'latitude': latitude,
      'category': category, //TODO: kiểm tra lại, chuyển từ ENUM -> String
      'description': description,
      'status': status, //TODO: kiểm tra lại, chuyển từ ENUM -> String
      //Relationship
      'userMember': json.decode(userMember!.toJson()),
      'userPartner': json.decode(userPartner!.toJson()),
      'ratingIssue': json.decode(ratingIssue!.toJson()),
      //
      'id': id,
      //'createdAt': createdAt,
      'createdBy': createdBy,
      //'updatedAt': updatedAt,
      'updatedBy': updatedBy,
      'version': version,
      'deleted': deleted,
    });
  }
}
